### PR TITLE
qa/workunits/rbd: minor rbd_mirror_helpers.sh fixes

### PR DESCRIPTION
**Pull request by nbalacha**
Corrected paths and added missing spaces.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
@tchaikov @cbodley Looks like those legacy-option-headers changes to CMake (which I became aware of due to incompatibility with older environments) don't capture all of the dependencies or are otherwise incomplete.  I haven't seen any failures related to the generated files before they landed, but have seen this happen at least twice recently:
```
[181/2743] No install step for 'rocksdb_ext'
[182/2743] Completed 'rocksdb_ext'
[183/2743] Generating mds-client.yaml
[184/2743] Generating mds-client_options.cc, ../../../include/mds-client_legacy_options.h
[185/2743] Generating mds_options.cc, ../../../include/mds_legacy_options.h
[186/2743] Generating mon.yaml
[187/2743] Generating osd.yaml
[188/2743] Generating mon_options.cc, ../../../include/mon_legacy_options.h
[189/2743] Generating rbd.yaml
[190/2743] Generating osd_options.cc, ../../../include/osd_legacy_options.h
[191/2743] Generating rbd-mirror.yaml
[192/2743] Generating rbd-mirror_options.cc, ../../../include/rbd-mirror_legacy_options.h
[193/2743] Generating immutable-object-cache.yaml
[194/2743] Generating rbd_options.cc, ../../../include/rbd_legacy_options.h
[195/2743] Generating ceph-exporter.yaml
[196/2743] Generating immutable-object-cache_options.cc, ../../../include/immutable-object-cache_legacy_options.h
[197/2743] Generating rgw.yaml
[198/2743] Generating ceph-exporter_options.cc, ../../../include/ceph-exporter_legacy_options.h
[199/2743] Creating directories for 'gtest-parallel_ext'
[200/2743] Building CXX object src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o
FAILED: src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o 
/usr/bin/ccache /usr/bin/clang++-14 -DBOOST_ASIO_DISABLE_THREAD_KEYWORD_EXTENSION -DBOOST_ASIO_HAS_IO_URING -DBOOST_ASIO_NO_TS_EXECUTORS -DHAVE_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_REENTRANT -D_THREAD_SAFE -D__CEPH__ -D__STDC_FORMAT_MACROS -D__linux__ -Dec_lrc_EXPORTS -I/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/include -I/home/jenkins-build/build/workspace/ceph-pull-requests/src -I/home/jenkins-build/build/workspace/ceph-pull-requests/src/erasure-code/jerasure -isystem /opt/ceph/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/build/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/jaegertracing/opentelemetry-cpp/api/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/jaegertracing/opentelemetry-cpp/ext/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/jaegertracing/opentelemetry-cpp/sdk/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/xxHash -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/fmt/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/erasure-code/jerasure/jerasure/include -isystem /home/jenkins-build/build/workspace/ceph-pull-requests/src/erasure-code/jerasure/gf-complete/include -g -Werror -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -Wall -fno-strict-aliasing -fsigned-char -Wtype-limits -Wignored-qualifiers -Wpointer-arith -Werror=format-security -Winit-self -Wno-unknown-pragmas -Wnon-virtual-dtor -Wno-ignored-qualifiers -ftemplate-depth-1024 -Wpessimizing-move -Wredundant-move -Wno-inconsistent-missing-override -Wno-mismatched-tags -Wno-unused-private-field -Wno-address-of-packed-member -Wno-unused-function -Wno-unused-local-typedef -Wno-varargs -Wno-gnu-designator -Wno-missing-braces -Wno-parentheses -Wno-deprecated-register -DCEPH_DEBUG_MUTEX -D_GLIBCXX_ASSERTIONS -fdiagnostics-color=auto -std=c++20 -MD -MT src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o -MF src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o.d -o src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o -c /home/jenkins-build/build/workspace/ceph-pull-requests/src/erasure-code/lrc/ErasureCodePluginLrc.cc
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/erasure-code/lrc/ErasureCodePluginLrc.cc:19:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/debug.h:18:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/dout.h:29:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/ceph_context.h:41:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/config_proxy.h:6:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/config.h:27:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/common/config_values.h:59:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/common/options/legacy_config_opts.h:10:10: fatal error: 'immutable-object-cache_legacy_options.h' file not found
#include "immutable-object-cache_legacy_options.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[201/2743] Generating rgw_options.cc, ../../../include/rgw_legacy_options.h
```

**Comment by idryomov** [comment]
jenkins test make check

**Comment by cbodley** [comment]
> @tchaikov @cbodley Looks like those legacy-option-headers changes to CMake (which I became aware of due to incompatibility with older environments) don't capture all of the dependencies or are otherwise incomplete. I haven't seen any failures related to the generated files before they landed, but have seen this happen at least twice recently:

thanks @idryomov. before Kefu's https://github.com/ceph/ceph/pull/57011, distro maintainers had complained about compilation failures due to ordering there (ex https://github.com/ceph/ceph/pull/56884). while i do think that pr was a step in the right direction, it was hard to validate fully because the failures are so dependent on build parallelism and timing

for better or worse, our ceph-common target is broken into a bunch of separate OBJECT libraries. these get bundled into ceph-common, but the object libraries themselves don't specify their own dependencies. modern cmake has gotten much better at treating these object libraries like normal targets (they carry transitive include directories, compile/link options, and link dependencies), so we just need to add the missing dependencies like Kefu did in https://github.com/ceph/ceph/pull/57583 and https://github.com/ceph/ceph/pull/57693

i audited each target from `grep -rw OBJECT --include=CMakeLists.txt` and added this dependency where i saw relevent includes like common/config.h or ceph_context.h etc. i pushed the result to https://github.com/ceph/ceph/pull/58414

> [200/2743] Building CXX object src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o
FAILED: src/erasure-code/lrc/CMakeFiles/ec_lrc.dir/ErasureCodePluginLrc.cc.o 

ec_lrc isn't itself an OBJECT library, but depends on some; this failure was addressed by adding that dependency to the crush_objs target. i was able to verify that by building the target, removing the generated headers from build/include/*options.h, and rebuilding the target

**Pull request merged**
